### ISSUE TITLE
Mask expansion over image's edges produces unwanted inpainting area

### ISSUE THREAD

**Issue by makkor-git** [lquesada/ComfyUI-Inpaint-CropAndStitch#61]
Hey there, I've been using your nodes for some time already, and they're really great! However, there's some behaviour I can't fully understand:

![Image](https://github.com/user-attachments/assets/c1ba6acd-ac3b-485d-a0ce-fdbb31839334)

As you can see, expanded mask goes way beyond image's edges, and expanded area is filled with "ones", making it an extra area for inpainting, which can even harm the inpainting process, messing with geometry and sizes.

I've discovered that you can avoid it by simply filling the new mask with zeros instead of ones in inpaint_cropandstitch,py, line 289:
`new_mask = torch.zeros((mask_batch, new_height, new_width), dtype=mask.dtype)`

![Image](https://github.com/user-attachments/assets/6a607652-ef8b-4f80-a8f9-0a20e6d23f40)

Or perhaps there is some drawback to my solution I am missing?

**Comment by lquesada** [comment]
hi,

This is working as intended: you're asking the node to force size to 1024x1024 (square) and extend the context 50% (1.5) around the mask. The only way to make this happen is by extending the mask beyond the original image.

The node prefills that extended area with "true" mask so as to avoid a weird (e.g. fully black) extended area influencing the sampling. This means the node will allow sampling outside the original image, but then the stitch node will get rid of that extra space.

**Comment by makkor-git** [comment]
> hi,
> 
> This is working as intended: you're asking the node to force size to 1024x1024 (square) and extend the context 50% (1.5) around the mask. The only way to make this happen is by extending the mask beyond the original image.
> 
> The node prefills that extended area with "true" mask so as to avoid a weird (e.g. fully black) extended area influencing the sampling. This means the node will allow sampling outside the original image, but then the stitch node will get rid of that extra space.

Yeah, I get the idea behind the reason for such behaviour and that stitch will get rid of all the extra space, but what do you mean by "to avoid a weird (e.g. fully black) extended area influencing the sampling"? How does the 'false' (black) mask influence the sampling?

In my tests, I've encountered cases when this extra inpainting area messed the overall geometry understanding (e.g. curved walls, etc.):

![Image](https://github.com/user-attachments/assets/c0ba17c6-568e-41f9-a744-af1f8add1d64)

**Comment by lquesada** [comment]
So... interestingly, using a zero mask around will mess it up even more - as what is in the extended image will deeply influence what's being generated - if it's all black image, those huge black borders will influence the masked image. If it's mirrored (as it is implemented now), not being able to regenerate over the mirrored area will influence the image.

The current one I believe is the best solution.
